### ISSUE TITLE
prop-shorthand: Add support for nested arrays

### ISSUE THREAD

**Issue by bitshadow** [wix/eslint-plugin-lodash#59]
```js
let fooArray;

fooArray = [{
    bar: [false],
    num: 1
}, {
    bar: [true],
    num: 2
}];
```

Method 1: 
```js
fooArray = _.filter(fooArray, (foo) => { // warning: Prefer property shorthand syntax
    return foo.bar[0];
});

console.log(fooArray);
```

Method 2:
```js
fooArray = _.filter(fooArray, ['foo.bar[0]', true]);

console.log(fooArray); // output = [] (empty array)
```

method1 shows warning message to use prop-shorthand syntax. While using shorthand syntax in method 2 it returns empty array.

**Comment by jdalton** [comment]
The path in Method 2 is incorrect. Your path `foo.bar[0]` is looking for `object.foo.bar[0]`. You just want:
```js
fooArray = _.filter(fooArray, ['bar[0]', true]);
```

**Comment by ganimomer** [comment]
What @jdalton said is correct. The rule works fine.

**Comment by bitshadow** [comment]
Thanks @jdalton. Works fine now. 👍

**Comment by bitshadow** [comment]
@jdalton now assuming below object. Is there any way to check for truthy (string) values? sorry I didn't see any example of this kind in doc so posting here.

```
let fooArray;

fooArray = [{
    bar: ["foostring"],
    num: 1
}, {
    bar: [false],
    num: 2
}];
```

I want to check if string is present or not. below example returns empty array in this case.
```
filterdArray = _.filter(fooArray, ['bar[0]']); // output: []
```

**Comment by ganimomer** [comment]
When you compare to a value, you can use either `_.matchesProperty` shorthand or `matches` shorthand.
The former (and what you asked in the first question) requires an array with a pair of path to value.
The latter requires an object (like the [docs for `_.matches`](https://lodash.com/docs#matches)

What you're trying to use is the `_.property` shorthand, which does **not** require an array.
That is:
```js
filteredArray = _.filter(fooArray, 'bar[0]'); // output: [{bar: ["foostring"], num: 1}]
```

Hope this helps!

**Comment by bitshadow** [comment]
Thanks @ganimomer for the clarification. That really helped.
